### DISCUSSION TITLE
EJBCA 8.0 with docker is not overriding custom config properties

### DISCUSSION THREAD

**Discussion by merryo11** [Keyfactor/ejbca-ce#358] [General]
While creating a new container in my docker environment through keyfactor/EJBCA-ce:latest image from github i am passing why own config properties files but my container is not taking the values from it and using the default properties values.

The command I am using is: (the command given is a sample and not the full command)

docker run -it -p 80:8080 -p 443:8443 --name node1 -h testing.com -e TLS_SETUP_ENABLED="true" -v /home/test/ejbca-custom-config/web.properties:/opt/keyfactor/ejbca/conf/web.properties keyfactor/ejbca-ce:latest 

I am passing three custom config files i.e. web,ejbca & install.properties in same above command but there is no effect.

Can any one give idea what wrong I am doing. Secondly while using PROXY_AJP_BIND and PROXY_HTTP_BIND with value as 0.0.0.0 through my environment variables (with -e option) my build gets stuck after some work and steps in some recursive loop and doesn't complete the process. I gave 24 hours for this process but it stucks in same loop

Thanks

**Comment by primetomas** [comment]
What are you trying to change? You should first look if there are any environment variables that will do what you want to do, i.e. look in:
https://hub.docker.com/r/keyfactor/ejbca-ce
use docker inspect to find environment variables. 

You should never need to modify bind port for PROXY_AJP_BIND or PROXY_HTT_BIND. I think that would be a sign that you are not following container best practices for ingress if you think you need to modify those.

**Reply by merryo11** [reply]
I am running below command exactly:

docker run -it -p 81:8080 -p 444:8443 --name pki-dev1 -h onsite-dev1.test.com  -e TLS_SETUP_ENABLED="true" -e PASSWORD_ENCRYPTION_KEY="12345" -e CA_KEYSTOREPASS="12345" -e EJBCA_CLI_DEFAULTPASSWORD="12345" -e HTTPSERVER_HOSTNAME="onsite-dev1.test.com" -e ADMINWEB_ACCESS="true" -e DATABASE_JDBC_URL="jdbc:mariadb://172.16.2.1:3306/pilotdb2?characterEncoding=UTF-8" -e DATABASE_USER="ejbca_dev" -e DATABASE_PASSWORD="12345678" -e ENABLE_DATABASE_SCHEMA_N_CONN_TEST="true" -e LOG_LEVEL_APP_WS_TRANSACTIONS="DEBUG" -e SMTP_DESTINATION="mail.test.com" -e SMTP_FROM="pki.alerts@test.com" -e SMTP_USERNAME="pki.alerts@test.com" -e SMTP_PASSWORD="12345" -e LOG_LEVEL_APP="DEBUG" -v /home/pkiuser/ejbca-cust-config/web.properties:/opt/keyfactor/ejbca/conf/web.properties -v /home/pkiuser/ejbca-cust-config/ejbca.properties:/opt/keyfactor/ejbca/conf/ejbca.properties -v /home/pkiuser/ejbca-cust-config/install.properties:/opt/keyfactor/ejbca/conf/install.properties keyfactor/ejbca-ce

**Reply by merryo11** [reply]
here are my custom properties files.

**Reply by merryo11** [reply]
[ejbca.properties.txt](https://github.com/Keyfactor/ejbca-ce/files/12715205/ejbca.properties.txt)
[web.properties.txt](https://github.com/Keyfactor/ejbca-ce/files/12715206/web.properties.txt)
[install.properties.txt](https://github.com/Keyfactor/ejbca-ce/files/12715207/install.properties.txt)

**Reply by primetomas** [reply]
I'm not sure what you actually want to change with these properties. Is it just the subjectDN of the Managment CA?
I.e. the different from the default values.

**Reply by primetomas** [reply]
Perhaps this deployment video showing Helm charts can be of help?
https://www.youtube.com/watch?v=C9dprcbCo4o

**Reply by svenska-primekey** [reply]
The install.properties is not used in the container and will not do anything. You should be using environment variables to configure settings in the web and ejbca properties files. The helm chart has the variables documented, https://github.com/Keyfactor/ejbca-community-helm.

**Reply by merryo11** [reply]
Yes I was targeting two things i.e. first, to change the ManagemantCA SubjectDN with my own SubjectDN and secondly I was trying to extend approval request time from 8 hours to 168 hours(1 Week) because of the use case so how can I do that.

For Management CA thing I may opt for creating a users under separate new CA chain and us it for RA purposes but for 2nd part how can I extend the timings.

**Comment by vacan1t** [comment]
Hey! Any updates on this? I am having the same issue trying to modify web.properties with a volume map` -v /opt/ejbca/properties/web.properties:/etc/ejbca/conf/web.properties:rw` in order to integrate with Fortanix HSM. The instructions says:
```
5. Add the following to the end of the web.properties file.
cryptotoken.p11.lib.60.name=Fortanix
cryptotoken.p11.lib.60.file=/opt/fortanix/pkcs11/fortanix_pkcs11.so
```

It seems like EJBCA can't read the new file because i get new warnings like this in the log after the change:
```
WARN  [org.ejbca.config.WebConfiguration] (default task-1) "web.docbaseuri" is not a valid URI. Using default value: internal
WARN  [org.ejbca.config.WebConfiguration] (default task-1) "httpserver.pubhttp" is not a decimal number. Using default value: 8080
WARN  [org.ejbca.config.WebConfiguration] (default task-1) "httpserver.privhttps" is not a decimal number. Using default value: 8443
```

**Reply by svenska-primekey** [reply]
Try adding a new layer to the container to update the web.properites. We have this documented over here: https://github.com/Keyfactor/keyfactorcommunity/tree/main/hsm-integration/fs-addon-fortanix
